internal/controllers/analyzer/analyzer.go

```diff
@@ -370,3 +392,37 @@ func (a *Analyzer) removeVulnerabilitiesByTypes() *analysis.Analysis {
 
 	return a.analysis
 }
+
+// setUpdateHashWarnings this func checks for hashes generated in older formats but that are still valid. If one of
+// these hashes are found, a warning will be showed informing the user to update the outdated hash.
+// TODO: Remove setUpdateHashWarnings before release v2.10.0
+// nolint
+func (a *Analyzer) setUpdateHashWarnings() {
+	isPrintDepreciationMsg := true
+
+	for vulnIndex := range a.analysis.AnalysisVulnerabilities {
+		for _, deprecatedHash := range a.analysis.AnalysisVulnerabilities[vulnIndex].Vulnerability.DeprecatedHashes {
```

Comment:
I think that would be good create a temp variable that holds `a.analysis.AnalysisVulnerabilities[vulnIndex]` value